fcrepo-kernel-modeshape/src/test/java/org/fcrepo/integration/kernel/modeshape/FedoraResourceImplIT.java

```diff
@@ -1221,8 +1197,10 @@ public void testGetNullNodeVersion() throws RepositoryException {
         final javax.jcr.Node node = getJcrNode(object);
         final Version version = node.getSession().getWorkspace().getVersionManager().getVersionHistory(node.getPath())
                 .getVersionByLabel("some-label");
-        final FedoraResource frozenResource = new FedoraResourceImpl(version.getFrozenNode().getNode("a"));
-        assertNull(frozenResource.getVersion("some-label"));
+
+        // Versioned child should not exist for versioning disabled child: PathNotFoundException Expected.
+        thrown.expect(PathNotFoundException.class);
+        version.getFrozenNode().getNode("a");
 
     }
 
```

Comment:
it might be good to have a test in here that shows the ldp:contains relationship still points to the canonical link of the child nodes?  As in, a test that shows the version of the resource still has it's ldp:contains intact.

Reply:
@bseeger I only see the current implementation with versioned child hierarchy snapshot has ldp:contains for the versioned child, so once the versioned child ignored, I don't see any ldp:contains links point to its child nodes. Maybe this is something we need to build.

Reply:
that being said, I'm not sure if it belongs here or added via some other PR later one.